lib/puma/server.rb

```diff
@@ -120,15 +120,15 @@ def current
       # :nodoc:
       # @version 5.0.0
       def tcp_cork_supported?
-        RbConfig::CONFIG['host_os'] =~ /linux/ &&
+        RbConfig::CONFIG['host_os'].include?('linux') &&
           Socket.const_defined?(:IPPROTO_TCP) &&
           Socket.const_defined?(:TCP_CORK)
       end
 
       # :nodoc:
       # @version 5.0.0
       def closed_socket_supported?
-        RbConfig::CONFIG['host_os'] =~ /linux/ &&
+        RbConfig::CONFIG['host_os'].include?('linux') &&
```

Comment:
What's the motivation for the change here?

Reply:
General code improvement.

1. If you need a boolean, choose a method that returns one.  I think that was the reason `String#match?` was added in Ruby 2.4.
2. Use string methods if a RegExp test isn't needed.  If and when macOS supports 'cork' (and hence `/linux|darwin/`), it can be a string method.

**Off topic:** A long time ago, my first 'significant' contribution to 'OS Ruby' was working on building & testing Windows/MinGW Ruby.  Since then, I've always kept an eye on the commits to ruby/ruby.  There are a lot of commits there that include small code changes like the change above.

I can revert...

Reply:
@nateberkopec

I looked at this a bit more, and `SOCKET::TCP_CORK` is not defined in Windows and macOS.  So, a better solution might be to add `HAS_TCP_CORK` to detect.rb and add a conditional to the method calls? So:
```ruby
cork_socket io
uncork_socket io
```

would change to:
```ruby
cork_socket(io) if HAS_TCP_SOCKET
uncork_socket(io) if HAS_TCP_SOCKET
```

Reply:
Ran `ruby -rsocket -e 'puts Socket.constants.sort'` in CI, and updated the code based on that.  The needed constants are not defined in Windows or macOS, so removed the linux detection...